src/core/data.jl

```diff
@@ -788,33 +806,35 @@ end
 ""
 function _check_cost_functions(id, comp)
     if "model" in keys(comp) && "cost" in keys(comp)
-        if comp["model"] == 1
-            if length(comp["cost"]) != 2*comp["ncost"]
-                error("ncost of $(comp["ncost"]) not consistent with $(length(comp["cost"])) cost values")
-            end
-            if length(comp["cost"]) < 4
-                error("cost includes $(comp["ncost"]) points, but at least two points are required")
-            end
-            for i in 3:2:length(comp["cost"])
-                if comp["cost"][i-2] >= comp["cost"][i]
-                    error("non-increasing x values in pwl cost model")
+        for ph in 1:length(comp["ncost"])
+            if comp["model"][ph] == 1
+                if length(PowerModels.getmpv(comp["cost"], ph)) != 2*comp["ncost"][ph]
```

Comment:
Note, inside of PowerModels you can call `getmpv` without needing to explicitly quantify it.